### ISSUE TITLE
Err:[cluster.go 597] no node set available for creating a data partition 

### ISSUE THREAD

**Issue by vnline** [chubaofs/chubaofs#489]
根据文档的指引配置部署一套测试集群：
Master: 3 nodes
metanode: 5 nodes 
datanode: 5 nodes
metanode 与 datanode节点混合部署，机型都是 128G RAM，6T * 3 HDD
创建了2个Vol，一个数据量：841G (capacity 100)，一个数据量：20.67T (capacity 1000)
创建了dataPartition ：4580
目前无法写入新数据，master日志提示：no node set available for creating a data partition 
没有在文档中找到相关描述，请问一下dataPartition 与 datanode之间的关系是怎样的，dataPartition可创建数量与哪些参数有关系？
另外，如果数据误删除如何恢复？是否有相关文档可以提供学习。
麻烦解答下，谢谢！

**Comment by zhuhyc** [comment]
@vnline  You can use the /topo/get API to show cluster topology information. ChubaoFS cluster topological structure is topo-zone-nodeset-node. When creating a partition, choose a available zone first, then choose a node set which must have 3(default replica number) writeable nodes called writebale node set. If no writeable node set,it will be to throw the error which is '_no node set available for creating a data partition_'.
    ChubaoFS does not support the recovery of deleted data. Thanks a lot.

**Comment by vnline** [comment]
`{
    "code": 0,
    "data": {
        "NodeSet": {
            "1": {
                "Cells": [
                    {
                        "DataNodes": [
                            {
                                "Addr": "10.129.51.199:17310",
                                "ID": 8,
                                "IsWritable": false,
                                "Status": true
                            },
                            {
                                "Addr": "10.129.51.198:17310",
                                "ID": 7,
                                "IsWritable": false,
                                "Status": true
                            },
                            {
                                "Addr": "10.129.51.200:17310",
                                "ID": 9,
                                "IsWritable": false,
                                "Status": true
                            },
                            {
                                "Addr": "10.129.51.204:17310",
                                "ID": 11,
                                "IsWritable": true,
                                "Status": true
                            },
                            {
                                "Addr": "10.129.51.201:17310",
                                "ID": 10,
                                "IsWritable": false,
                                "Status": true
                            }
                        ],
                        "Name": "default"
                    }
                ],
                "MetaNodes": [
                    {
                        "Addr": "10.129.51.204:17210",
                        "ID": 6,
                        "IsWritable": true,
                        "Status": true
                    },
                    {
                        "Addr": "10.129.51.200:17210",
                        "ID": 5,
                        "IsWritable": true,
                        "Status": true
                    },
                    {
                        "Addr": "10.129.51.199:17210",
                        "ID": 3,
                        "IsWritable": true,
                        "Status": true
                    },
                    {
                        "Addr": "10.129.51.198:17210",
                        "ID": 2,
                        "IsWritable": true,
                        "Status": true
                    },
                    {
                        "Addr": "10.129.51.201:17210",
                        "ID": 4,
                        "IsWritable": true,
                        "Status": true
                    }
                ]
            }
        }
    },
    "msg": "success"
}`
从topo可以确认只有一个NodeSet，是否可以对现存数据进行副本数或topolopy进行调整，实现4个节点存储量可用最大化设计？
部署时如何配置topolopy扩展信息的？貌似这块没看到有具体说明。

**Comment by zcalusic** [comment]
I've followed the instructions on https://chubaofs.readthedocs.io/en/latest/manual-deploy.html page, just to arrive to this same error. Whatever I do, no partitions can be created, file system can't be mounted etc... That /topo/get call is not mentioned on the deployment page, though I managed to get the information and all three datanodes have isWritable set to false. I don't know why, and I don't know how to proceed from here...

I guess I will stop evaluating the (otherwise quite promising) software now, until the times the documentation is improved. Right now it's seriously lacking... :(
